src/lib.rs

```diff
@@ -208,12 +208,22 @@ pub struct Tag<'a> {
 struct Record {
     freq: usize,
     tag: String,
+    deleted: bool,
 }
 
 impl Record {
     #[inline(always)]
     fn new(freq: usize, tag: String) -> Self {
-        Self { freq, tag }
+        Self {
+            freq,
+            tag,
+            deleted: false,
+        }
+    }
+
+    fn mark_deleted(&mut self) {
+        self.deleted = true;
```

Comment:
Didn't see any code check this flag?

Reply:
just `mark_deleted` for soft delete which means this `word_id` has been used. see `remove_word`.

If a large number of words (more than half) are deleted, then consider using the deleted positions in `add_words`, which will sacrifice performance. But I think the use case of `remove_word` is a small change. So I choose to waste a little space and maintain performance.

Reply:
Moreover, maybe sacrificing the performance of `add_word` is acceptable to prevent users from using `remove_word` a lot.